src/Microsoft.Extensions.HealthChecks.Abstractions/Microsoft.Extensions.HealthChecks.Abstractions.csproj

```diff
@@ -2,6 +2,9 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <RootNamespace>Microsoft.Extensions.HealthChecks</RootNamespace>
+    <LangVersion>latest</LangVersion>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
I'm no build expert but don't we usually get these from some kind of msbuild thing rather than in the project directly?

Reply:
Yep, this is just until we're inside the aspnet org.

Reply:
I kinda want us to put all of this new stuff in a single repo but we can't agree on what to call it. The aussies hate all my ideas for what it would be called

Reply:
I guess the question is "what new stuff?". I think this fits in along with things like Metrics and other Telemetry abstractions (like Logging), which is why my preference is Logging (possibly with a rename).

Reply:
Diagnostics is a better place

Reply:
> But make sure you remember that it isn't just for ASP.NET Core.

Yep, the current code is all `Microsoft.Extensions.` because the middleware is out of scope for this first PR.

> Are you also suggesting that it is Microsoft.Extensions.Diagnostics.HealthChecks

That I'm less sure of, but I do think it makes some sense to put these under a Diagnostics root.

Reply:
Lets talk about it today if we can assemble all of the ivory fishbowl folks.